safehttp/response.go

```diff
@@ -41,11 +43,32 @@ type Template interface {
 // TemplateResponse bundles a Template with its data and names to function
 // mappings to be passed together to the commit phase.
 type TemplateResponse struct {
-	Template *Template
-	Data     *interface{}
+	Template Template
+	Data     interface{}
 	FuncMap  map[string]interface{}
 }
 
 // NoContentResponse is sent to the commit phase when it's initiated from
 // ResponseWriter.NoContent.
 type NoContentResponse struct{}
+
+// JSONResp creates a JSONResponse from a data object. This should be a
+// valid JSON object that can be serialised and written to the
+// http.ResponseWriter using a JSON encoder.
+func JSONResp(data interface{}) JSONResponse {
+	return JSONResponse{Data: data}
+}
+
+// TemplateResp creates a TemplateResponse from a Template, its data and
+// the name to function mappings. These will be pre-processed and then written
+// to the http.ResponseWriter, if deemed safe. If the funcMap is non-nil, its
+// elements will override the existing name to function mappings in the
+// template. An attempt to define a new name to function mapping that is not
+// already in the template will result in a panic.
+func TemplateResp(t Template, data interface{}, funcMap map[string]interface{}) TemplateResponse {
+	return TemplateResponse{
+		Template: t,
+		Data:     data,
+		FuncMap:  funcMap,
+	}
+}
```

Comment:
Let's leave it as it is for now, but I'm not a fan of `FooResp`, especially since it clashes with `FooResponse` and whoever uses autocompletion will get confused.

I suppose we should revisit the idea from: https://github.com/google/go-safeweb/pull/205#discussion_r504652946.

cc @empijei

Reply:
I think it's not a big change to add the two high-level functions.